src/trans/gpu/CMakeLists.txt

```diff
@@ -0,0 +1,197 @@
+# (C) Copyright 2020- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+## Apply workarounds for some known compilers
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "NVHPC")
+
+  # Fix for IFS "CONGRAD: SPTSV/DPTSV returned non-zero info with crayftn 8.7.7 (cdt/18.12)
+  ectrans_add_compile_options(
+      SOURCES external/setup_trans.F90
+      FLAGS   "-fast -O3 -acc -gpu=pinned,deepcopy,fastmath,nordc")
+
+endif()
+
+## Assemble sources
+
+ecbuild_list_add_pattern( LIST trans_src
+                          GLOB
+                                sharedmem/*
+                                algor/*
+                                internal/*
+                                external/*
+                          QUIET
+                        )
+
+
+## for reduced memory option, remove duplicate source files
+if( HAVE_REDUCED_MEMORY )
+  ecbuild_list_add_pattern( LIST reducedmem_files
+                            GLOB internal_reducedmem/* QUIET )
+  ecbuild_list_add_pattern( LIST trans_src
+	                    GLOB internal_reducedmem/* QUIET )
+  foreach( src_file ${reducedmem_files} )
+    get_filename_component( base_name ${src_file} NAME )
+    list(REMOVE_ITEM trans_src "internal/${base_name}" )
+  endforeach()
+endif()
+
+ecbuild_list_exclude_pattern( LIST trans_src REGEX dilatation_mod.F90 )
+ecbuild_list_exclude_pattern( LIST trans_src_reducedmem REGEX dilatation_mod.F90 )
+if( NOT HAVE_FFTW )
+  ecbuild_list_exclude_pattern( LIST trans_src REGEX tpm_fftw.F90 )
+  ecbuild_list_exclude_pattern( LIST trans_src_reducedmem REGEX tpm_fftw.F90 )
+endif()
+
+
+set( HAVE_dp ${HAVE_DOUBLE_PRECISION} )
+set( HAVE_sp ${HAVE_SINGLE_PRECISION} )
```

Comment:
Above 2 lines can be deleted. They are already present in outmost scope. I had moved them there because they are repeated in many places